php/utility/fileutil.php

```diff
@@ -64,12 +63,12 @@ public static function getProfilePath( $user = null )
 		return(self::$profilePathInstance);
 	}
 	
-	public static function getSettingsPath( $user = null )
+	public static function getSettingsPath()
 	{
 		return( self::getProfilePath($user).'/settings' );
```

Comment:
1) This is incorrect, because now getProfilePath does not have parameters.
2) Please, keep '$user' parameter for getUploadsPath/getSettingsPath/getProfilePath functions. This is required for the third-party plugins compatibility.

Reply:
Third party plugins are going to break anyways by this pull request. Are you okay with a new set of functions for this functionality? It's not possible to cache the profile path for the user, with this '$user' parameter.

**For instance:** getUploadsPathEx/getSettingsPathEx/getProfilePathEx could have the user parameter and no caching.

Reply:
> It's not possible to cache the profile path for the user, with this '$user' parameter.

Why? For example, something like that:

```
if(!isset(self::$profilePathInstance[$user]))
{
	self::$profilePathInstance[$user] = ...
}
return(self::$profilePathInstance[$user]);
```

But this is not principal.